Upgrade Introduction chapter, include Schema (#72)
r-devel/rdevguide: 01-Introduction.Rmd

# Introduction

This guide is heavily influenced by the [Python Developer Guide](https://devguide.python.org/), and is a comprehensive resource for contributing to base R – for both new and experienced contributors. It is maintained by the [R Contribution Working Group](/working-group). We welcome your contributions to base R!

> ## Introduction to Bugzilla

Reviewer: I don't think the introduction to Bugzilla fits in this chapter as it is only relevant to people working on bugs - it may not even be necessary for people that only want to report a bug as they could do that on the R-devel mailing list.

I suggest moving this section to the reviewing bugs chapter, as the bug tracking chapter is already quite long and has direct links to Bugzilla where necessary (e.g. in item 4 in this section: https://contributor.r-project.org/rdevguide/bug-tracking.html#RCorePkgBug). I think we could add a Quick Overview of Bugzilla as a subsection before or after this one: https://contributor.r-project.org/rdevguide/reviewing-bugs.html#preparing-to-review-bug-reports, since getting familiar with the different search options is probably most helpful to people that want to contribute to bug review. We could add a link to this Quick Overview of Bugzilla in the Bug Tracking chapter as well (ch 3) to help people navigate Bugzilla without having to come back to the direct links. (Navigating Bugzilla could be an alternative title for the subsection).

<!-- TODO(Saranjeet): Add other operating systems as they get included in the guide-->

2. The [Bug Tracking](#BugTrack) and the [Reviewing Bugs](#ReviewBugs) chapters discuss how to find bugs in R and how to review bug reports that are submitted to Bugzilla.

3. The [Finding the Source](#FindSource) chapter discusses how to have an overview of the R codebase like finding the implementation of a base function written in R and/or the primitive implementation written in C.

4. The [Lifecycle of a Patch](#FixBug) chapter discusses how to create a patch to propose a bug fix.

5. The [Documenting R](#Doc) chapter describes the styleguide for R's documentation, how to document for R, and report and review bugs (and/or suggest corrections for them) in the existing documentation.

6. How to test pre-release versions of R is discussed in the chapter on [Testing Pre-release R Versions](#TestRVer).

7. The [R Core Developers](#CoreDev) chapter contains a list of the former and current members of the R Core team who have write access to the R source.

> 8. To get more information on how to find assistance to resolve issues faced while working on R, refer to the [Where to Get Help](#WhereToGetHelp) chapter.

Reviewer 2: ```suggestion
8. For more information on how to engage with to community and ask for help, refer to the [Where to Get Help](#WhereToGetHelp) chapter.
```

9. To keep up with the developments in R refer to some of the resources available in the [News and Announcements](#News)

10. Resources and tools that may be relevant for R developers are available in the [Developer Tools](#DevTools) chapter.

## Overview of different ways of contributing to base R Project

> One can contribute to the base R project in a number of different ways. Some of them are listed below:

Reviewer 2: ```suggestion
Contributions to base R are possible in a number of different ways. Some of them are listed below:
```

Reviewer: I think this overview should come much earlier, after the introductory paragraph. This is what new contributors need to know right at the start: the main ways that the can contribute to the R project and which chapter to jump to if they want to get started on a particular topic.

Then you could have the more detailed overview of every chapter (for those that want to find out more about the big picture before jumping in).
